### PR TITLE
KAFKA-16331: remove EOSv1 config from StreamsConfig

### PR DESCRIPTION
**Pull request by mjsax**
Stacked on #17110 

This PR add a couple of workaround (marked as TODO) to limit the scope. TODOs will be resolved via follow up PRs.

### CODE REVIEW

File: streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java [hunk 1]
```diff
@@ -1775,11 +1735,7 @@ public Map<String, Object> getProducerConfigs(final String clientId) {
         props.putAll(getClientCustomProps());
         props.putAll(clientProvidedProps);
 
-        // When using EOS alpha, stream should auto-downgrade the transactional commit protocol to be compatible with older brokers.
-        if (StreamsConfigUtils.processingMode(this) == StreamsConfigUtils.ProcessingMode.EXACTLY_ONCE_ALPHA) {
-            props.put("internal.auto.downgrade.txn.commit", true);
-        }
-
+        // TODO: given that we remove this code, it seems we can remove the corresponding tests (cf other TODO)
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java]: Open question (we can resolve in this PR). Refers to removing the code above which sets `internal.auto.downgrade.txn.commit`.

File: streams/src/main/java/org/apache/kafka/streams/internals/StreamsConfigUtils.java [hunk 2]
```diff
@@ -42,12 +42,12 @@ public enum ProcessingMode {
             this.name = name;
         }
     }
-    
-    @SuppressWarnings("deprecation")
+
+    // TODO cleanup
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/internals/StreamsConfigUtils.java:46]: Will be addressed in a follow up PR -- for now, just replacing removed variables with hard-coded strings below.

File: streams/src/main/java/org/apache/kafka/streams/internals/StreamsConfigUtils.java [hunk 3]
```diff
@@ -56,12 +56,12 @@ public static ProcessingMode processingMode(final StreamsConfig config) {
         }
     }
 
-    @SuppressWarnings("deprecation")
+    // TODO cleanup
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/internals/StreamsConfigUtils.java:59]: Same -- follow up PR will refactor this further and remove hard coded string replacements.

File: streams/src/main/java/org/apache/kafka/streams/internals/StreamsConfigUtils.java [hunk 4]
```diff
@@ -28,11 +28,11 @@ public class StreamsConfigUtils {
 
     private static final Logger LOG = LoggerFactory.getLogger(StreamsConfigUtils.class);
 
-    @SuppressWarnings("deprecation")
     public enum ProcessingMode {
         AT_LEAST_ONCE(StreamsConfig.AT_LEAST_ONCE),
 
-        EXACTLY_ONCE_ALPHA(StreamsConfig.EXACTLY_ONCE),
+        // TODO cleanup
+        EXACTLY_ONCE_ALPHA("exactly_once"),
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/internals/StreamsConfigUtils.java:35]: Will be addressed in a follow up refactoring PR. Just a workaround to keep the scope contained.

File: streams/src/test/java/org/apache/kafka/streams/StreamsConfigTest.java [hunk 5]
```diff
@@ -600,24 +562,7 @@ public void shouldNotSetInternalAutoDowngradeTxnCommitToTrueInProducerForEosDisa
...
-    @SuppressWarnings("deprecation")
-    @Test
-    public void shouldNotSetInternalAutoDowngradeTxnCommitToTrueInProducerForEosBeta() {
-        props.put(StreamsConfig.PROCESSING_GUARANTEE_CONFIG, EXACTLY_ONCE_BETA);
-        final StreamsConfig streamsConfig = new StreamsConfig(props);
-        final Map<String, Object> producerConfigs = streamsConfig.getProducerConfigs(clientId);
-        assertThat(producerConfigs.get("internal.auto.downgrade.txn.commit"), is(nullValue()));
-    }
-
+    // TODO: should we keep this test? (cf other TODO)
```
Comment (mjsax) [inline streams/src/test/java/org/apache/kafka/streams/StreamsConfigTest.java]: Relates to the question above, about the remove code which sets the internal config.

File: streams/src/test/java/org/apache/kafka/streams/processor/internals/TaskManagerTest.java [hunk 6]
```diff
@@ -1578,6 +1574,10 @@ public void shouldRethrowTaskCorruptedExceptionFromStateUpdater() {
         assertEquals(mkSet(taskId00, taskId01), thrown.corruptedTasks());
         assertEquals("Tasks [0_1, 0_0] are corrupted and hence need to be re-initialized", thrown.getMessage());
     }
+
+    // this @Test annotation was removed via https://github.com/apache/kafka/pull/15896
+    // but I am not sure why -- maybe by mistake?
+    @Test
```
Comment (mjsax) [inline streams/src/test/java/org/apache/kafka/streams/processor/internals/TaskManagerTest.java:1669]: Does anybody know? @cadonna maybe?

Comment (cadonna) [reply streams/src/test/java/org/apache/kafka/streams/processor/internals/TaskManagerTest.java:1669]: I am quite sure, it was a mistake that we missed.

File: streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java (hunk 1 above)
Comment (bbejeck) [reply streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java]: My vote is to do the cleanup in this PR and remove `internal.auto.downgrade.txn.commit` and associated code

File: streams/src/main/java/org/apache/kafka/streams/internals/StreamsConfigUtils.java [hunk 7]
```diff
@@ -42,12 +42,12 @@ public enum ProcessingMode {
...
         }
     }
-    
-    @SuppressWarnings("deprecation")
+
+    // TODO cleanup
     public static ProcessingMode processingMode(final StreamsConfig config) {
-        if (StreamsConfig.EXACTLY_ONCE.equals(config.getString(StreamsConfig.PROCESSING_GUARANTEE_CONFIG))) {
+        if ("exactly_once".equals(config.getString(StreamsConfig.PROCESSING_GUARANTEE_CONFIG))) {
             return ProcessingMode.EXACTLY_ONCE_ALPHA;
```
Comment (bbejeck) [inline streams/src/main/java/org/apache/kafka/streams/internals/StreamsConfigUtils.java:49]: maybe take the same approach with `EXACTLY_ONCE_ALPHA` ? Same for the other occurances

File: streams/src/test/java/org/apache/kafka/streams/StreamsConfigTest.java (hunk 5 above)
Comment (bbejeck) [reply streams/src/test/java/org/apache/kafka/streams/StreamsConfigTest.java]: Yes, let's remove in this PR

File: streams/src/test/java/org/apache/kafka/streams/integration/EosIntegrationTest.java [hunk 8]
```diff
@@ -176,13 +175,13 @@ public void createTopics() throws Exception {
     }
 
     @ParameterizedTest
-    @ValueSource(strings = {StreamsConfig.AT_LEAST_ONCE, StreamsConfig.EXACTLY_ONCE, StreamsConfig.EXACTLY_ONCE_V2})
+    @ValueSource(strings = {StreamsConfig.AT_LEAST_ONCE, StreamsConfig.EXACTLY_ONCE_V2})
```
Comment (bbejeck) [inline streams/src/test/java/org/apache/kafka/streams/integration/EosIntegrationTest.java]: We don't have to address in this PR, but why do we have `AT_LEAST_ONCE` parameters in the EOS integration test?

**Review by bbejeck — COMMENTED** [review]
Thanks for PR @mjsax, overall this lgtm, just left a couple of comments

File: streams/src/main/java/org/apache/kafka/streams/internals/StreamsConfigUtils.java (hunk 7 above)
Comment (mjsax) [reply streams/src/main/java/org/apache/kafka/streams/internals/StreamsConfigUtils.java:49]: Yes, will cleanup `ProcessingMode`  in a follow up PR an remove it.

File: streams/src/test/java/org/apache/kafka/streams/integration/EosIntegrationTest.java (hunk 8 above)
Comment (mjsax) [reply streams/src/test/java/org/apache/kafka/streams/integration/EosIntegrationTest.java]: As discussed on the other PRs... need to clean this up. Filed https://issues.apache.org/jira/browse/KAFKA-17558

**Review by bbejeck — APPROVED** [review]
LGTM

**Comment by bbejeck** [comment]
@mjsax looks like we have a single related test failure `StreamsConfigTest.testCaseInsensitiveSecurityProtocol()`

File: streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamsProducerTest.java [hunk 9]
```diff
@@ -102,11 +101,11 @@ public class StreamsProducerTest {
         mkEntry(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, "dummy:1234"))
     );
 
-    @SuppressWarnings("deprecation")
+    // TODO cleanup
```
Comment (mjsax) [inline streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamsProducerTest.java:104]: Addressed via https://github.com/apache/kafka/pull/17212

**Pull request merged**